File: Sc.Models/Dtos/Suppliers/Search/SupplierSearchDto.cs
using Sc.Models.Entities.Base;

namespace Sc.Models.Dtos.Suppliers.Search
{
    public class SupplierSearchDto : Entity
    {
        // Common
        public string Name { get; set; }
        public string NameAlt { get; set; }

        // Institution
        public string Uic { get; set; }
        public int? RootId { get; set; }
        public string RootName { get; set; }
        public string RootNameAlt { get; set; }

        public List<SupplierOfferingDto> SupplierOfferings { get; set; } = new List<SupplierOfferingDto>();
    }
}

File: WebApplication/src/suppliers/dtos/search/supplier-search.dto.ts
import { SupplierOfferingDto } from "../supplier-offering.dto";

export class SupplierSearchDto {
    id: number;

    // Common
    name: string;
    nameAlt: string;

    // Institution
    uic: string;
    rootId: number;
    rootName: string;
    rootNameAlt: string;

    supplierOfferings: SupplierOfferingDto[] = [];
}